### ISSUE TITLE
`leading_backslash_in_global_namespace` adds backslash to native types in phpdoc in global namespace

### ISSUE THREAD

**Issue by Riimu** [PHP-CS-Fixer/PHP-CS-Fixer#7695]
### Description

The fixer `fully_qualified_strict_types` with `leading_backslash_in_global_namespace` option set to `true` will prefix native types with backslashes in phpdoc in global namespace, which should not happen.

### Runtime version

```
➜  type-bug-fixer vendor/bin/php-cs-fixer -V
PHP CS Fixer 3.46.0 Three Keys by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.2.14
```

### Used command

```
➜  type-bug-fixer vendor/bin/php-cs-fixer fix -vvv --dry-run --diff
```

### Configuration file

```php
<?php

$finder = (new PhpCsFixer\Finder())
    ->in(__DIR__)
;

return (new PhpCsFixer\Config())
    ->setRules([
        'fully_qualified_strict_types' => ['leading_backslash_in_global_namespace' => true]
    ])
    ->setFinder($finder)
;
```

### Code snippet that reproduces the problem

The file `test.php`

```php
<?php

/**
 * @param int $hello
 */
function foo (int $hello): void {}
```

Result of running php-cs-fixer:

```
PHP CS Fixer 3.46.0 Three Keys by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.2.14
Loaded config default from "/Users/riikka/Projects/type-bug-fixer/.php-cs-fixer.php".
Using cache file ".php-cs-fixer.cache".
 1/1 [████████████████████████████] 100%

   1) test.php (fully_qualified_strict_types)
      ---------- begin diff ----------
--- /Users/riikka/Projects/type-bug-fixer/test.php
+++ /Users/riikka/Projects/type-bug-fixer/test.php
@@ -1,6 +1,6 @@
 <?php

 /**
- * @param int $hello
+ * @param \int $hello
  */
 function foo (int $hello): void {}

      ----------- end diff -----------


Found 1 of 1 files that can be fixed in 0.008 seconds, 14.000 MB memory used
```

Expected no fixes to be made

**Comment by Wirone** [comment]
I believe it's fixed in #7648 but not released yet. @Riimu Can you verify using `master` branch?

**Comment by Riimu** [comment]
So it appears. Apologies for not realizing to test against master.

```
➜  type-bug-fixer vendor/bin/php-cs-fixer fix -vvv --dry-run --diff
PHP CS Fixer 3.46.1-DEV Three Keys by Fabien Potencier, Dariusz Ruminski and contributors.
PHP runtime: 8.2.14
Loaded config default from "/Users/riikka/Projects/type-bug-fixer/.php-cs-fixer.php".
Using cache file ".php-cs-fixer.cache".
 1/1 [████████████████████████████] 100%


Found 0 of 1 files that can be fixed in 0.003 seconds, 14.000 MB memory used
```
